Apps/TimelineApp/feature/user/src/main/java/uk/ac/cam/cares/jps/user/SensorSettingFragment.java

```diff
@@ -0,0 +1,297 @@
+package uk.ac.cam.cares.jps.user;
+
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.Toast;
+
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.Fragment;
+import androidx.lifecycle.ViewModelProvider;
+import androidx.navigation.fragment.NavHostFragment;
+import androidx.recyclerview.widget.DividerItemDecoration;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.google.android.material.dialog.MaterialAlertDialogBuilder;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import dagger.hilt.android.AndroidEntryPoint;
+import uk.ac.cam.cares.jps.sensor.source.handler.AccelerometerHandler;
+import uk.ac.cam.cares.jps.sensor.source.handler.GravitySensorHandler;
+import uk.ac.cam.cares.jps.sensor.source.handler.GyroscopeHandler;
+import uk.ac.cam.cares.jps.sensor.source.handler.LightSensorHandler;
+import uk.ac.cam.cares.jps.sensor.source.handler.LocationHandler;
+import uk.ac.cam.cares.jps.sensor.source.handler.MagnetometerHandler;
+import uk.ac.cam.cares.jps.sensor.source.handler.PressureSensorHandler;
+import uk.ac.cam.cares.jps.sensor.source.handler.RelativeHumiditySensorHandler;
+import uk.ac.cam.cares.jps.sensor.source.handler.SensorManager;
+import uk.ac.cam.cares.jps.sensor.source.handler.SensorType;
+import uk.ac.cam.cares.jps.sensor.source.handler.SoundLevelHandler;
+import uk.ac.cam.cares.jps.user.databinding.FragmentSensorSettingBinding;
+import uk.ac.cam.cares.jps.user.viewmodel.AccountViewModel;
+import uk.ac.cam.cares.jps.user.viewmodel.SensorAdapter;
+import uk.ac.cam.cares.jps.user.viewmodel.SensorViewModel;
+
+/**
+ * Fragment that manages the sensor settings UI and handles user interactions
+ * related to starting and stopping sensor recordings.
+ */
+@AndroidEntryPoint
+public class SensorSettingFragment extends Fragment implements OnSensorToggleListener {
+
+    private FragmentSensorSettingBinding binding;
+    private SensorViewModel sensorViewModel;
+    private AccountViewModel accountViewModel;
+    private Map<Permission.PermissionType, Permission> permissionsMap = new HashMap<>();
+    private List<Permission.PermissionType> criticalPermissionType = Arrays.asList(Permission.PermissionType.LOCATION_FINE, Permission.PermissionType.AUDIO);
+    @Inject
+    SensorManager sensorManager;
+    private SensorAdapter adapter;
+    private boolean allToggledOn = false;
```

Comment:
the fragment is an UI level component and should not host any states. States should be managed by ViewModel.